main.tf

```diff
@@ -0,0 +1,105 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      version = ">= 2.49"
+    } 
+  }
+  backend "azurerm" {
+        resource_group_name  = "tfstate"
+        storage_account_name = "tfstate13267"
+        container_name       = "tfstate"
+        key                  = "terraform.tfstate"
+    }
+}
+provider "azurerm" {
+  features {
+
+  }
+}
+
+data "azurerm_resource_group" "main" {
+  name     = "devops-todo-app"
+}
+
+resource "azurerm_app_service_plan" "main" {
+  name                = "todoapp-jaigeneric-plan"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  kind                = "Linux"
+  reserved            = true
+  sku {
+    tier = "Basic"
+    size = "B1"
+  } 
+}
+
+resource "azurerm_app_service" "main" {
+  name                = "todoapp-jaigeneric"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  app_service_plan_id = azurerm_app_service_plan.main.id
+  site_config {
+    app_command_line = ""
+    linux_fx_version = "DOCKER|jaisalpatel836/todo_app:latest"
+  }
+  app_settings = {
+    "DOCKER_REGISTRY_SERVER_URL" = "https://index.docker.io"
+    "mongo_db_connection" = resource.azurerm_cosmosdb_account.main.connection_strings[0]
+    "mongo_db_name" = "todo_app"
+    "SECRET_KEY" = "secret-key"
```

Comment:
Minor here but if this was a real project it would be very important that this is a variable that is sensitive and managed as such; as it is I'm fairly confident I genuinely could bypass your security by directly adjusting my session cookie, but I appreciate through the course we've treated this variable fairly insensitively!